pkg/util/workqueue/queue.go

```diff
@@ -113,6 +113,17 @@ func (q *Type) Get() (item interface{}, shutdown bool) {
 	return item, false
 }
 
+// Get the head of the queue
+func (q *Type) Peek() interface{} {
```

Comment:
IMO peek is a code smell, super easy to use it wrong in multithreaded environments. I would prefer not to add this unless there's no other way around it. E.g., you could make Add() return true if it actually added to the queue.